server/context.go

```diff
@@ -0,0 +1,168 @@
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Daniel Theophanes (kardianos@gmail.com)
+
+package server
+
+import (
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/storage/engine"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+// Context holds parameters needed to setup a server.
+type Context struct {
+	// RPC is the "host:port" to bind for RPC traffic.
+	RPC string
+
+	// HTTP is the "host:port to bind for HTTP traffic.
+	HTTP string
+
+	// Addr is the address for connection to cockroach cluster.
+	Addr string
+
+	// Certs specifies a directory containing RSA key and x509 certs.
+	Certs string
+
+	// Stores is specified to enable durable key-value storage.
+	// Memory-backed key value stores may be optionally specified
+	// via mem=<integer byte size>.
+	//
+	// Stores specify a comma-separated list of stores specified by a
+	// colon-separated list of device attributes followed by '=' and
+	// either a filepath for a persistent store or an integer size in bytes for an
+	// in-memory store. Device attributes typically include whether the store is
+	// flash (ssd), spinny disk (hdd), fusion-io (fio), in-memory (mem); device
+	// attributes might also include speeds and other specs (7200rpm, 200kiops, etc.).
+	// For example, -store=hdd:7200rpm=/mnt/hda1,ssd=/mnt/ssd01,ssd=/mnt/ssd02,mem=1073741824
+	Stores string
+
+	// Attrs specifies node topography or machine capabilities, used to
+	// match capabilities or location preferences specified in zone configs.
+	Attrs string
+
+	// Maximum clock offset for the cluster.
+	MaxOffset time.Duration
+
+	// BootstrapOnly signals to only bootstrap and not start the server.
+	BootstrapOnly bool
+
+	// GossipBootstrap is a comma-separated list of node addresses that
+	// act as bootstrap hosts for connecting to the gossip network.
+	GossipBootstrap string
+
+	// GossipInterval is a time interval specifying how often gossip is
+	// communicated between hosts on the gossip network.
+	GossipInterval time.Duration
+
+	// Enables linearizable behaviour of operations on this node by making sure
+	// that no commit timestamp is reported back to the client until all other
+	// node clocks have necessarily passed it.
+	Linearizable bool
+
+	// CacheSize is the amount of memory in bytes to use for caching data.
+	// The value is split evenly between the stores if there are more than one.
+	CacheSize int64
+
+	// Parsed values.
+
+	// Engines is the storage instances specified by Stores.
+	Engines []engine.Engine
+
+	// NodeAttributes is the parsed representation of Attrs.
+	NodeAttributes proto.Attributes
+}
+
+// NewContext returns a Context with default values.
+func NewContext() *Context {
+	return &Context{
+		RPC:  ":0",
```

Comment:
I think you should break all of these values out into a const() section at top of file.